src/static_canvas.class.js

```diff
@@ -868,10 +864,17 @@
       if (this.clipTo) {
         fabric.util.clipContext(this, canvasToDrawOn);
       }
+      
+      objsToRender = this._chooseObjectsToRender();
```

Comment:
Following your advice i grouped the big block of code and gave it a clear ( i hope ) name.